File: LeetCode/Binary_Search/official/268_missing_number/268_missing_number.java
class Solution {
    public static int missingNumber(int[] nums) {
        	int size = nums.length;
	        int res = size * (size + 1) / 2; // the math formula
        	int sum = 0;

	        for (int n: nums)
        	    sum += n;
	        return (res - sum);
	}
    /*public static void main(String[] args) {
		int[] nums = {3, 0, 1};
		System.out.println(Solution.missingNumber(nums));
	}*/
}

File: LeetCode/Binary_Search/official/268_missing_number/268_missing_number.py
class Solution(object):
    def missingNumber(self, nums):
       size = len(nums)
       res = size * (size + 1) // 2
       total = 0

       for n in nums:
            total += n
       return res - total  

'''
if __name__ == "__main__":
	sol = Solution()
	nums = [3, 0 ,1]
	result = sol.missingNumber(nums) 
	print(result)
'''
